tests/integration/high_availability/test_primary_switchover.py

```diff
@@ -56,13 +58,30 @@ def test_cluster_failover_after_majority_loss(juju: Juju, highly_available_clust
 
     logging.info(f"Unit selected for promotion: {unit_to_promote}")
 
-    logging.info("Rebooting all but one unit to simulate majority loss...")
-    for unit in [non_primary_units.pop(), primary_unit]:
-        machine_name = get_unit_machine(juju, app_name, unit)
-        run(["lxc", "restart", machine_name], check=True)
+    logging.info("Kill all but one unit to simulate majority loss...")
+    units_to_kill = [non_primary_units.pop(), primary_unit]
+    machine_name = []
+    for unit in units_to_kill:
+        machine_name.append(get_unit_machine(juju, app_name, unit))
+
+    run(["lxc", "restart", "--force", machine_name[0], machine_name[1]], check=True)
```

Comment:
Shouldn't this rather be `stop`? Wouldn't the restarted units eventually go live and rejoin?

Reply:
Because we force stop 2 (out of 3) simultaneously, there will be no quorum for reelecting a new primary since only one remained. At the time the 2 restart ones come back up, they can rejoin if there's no primary. And that's exactly the scenario we need to simulate here.
One could also just recover the remaining unit, and start others after, but we test more doing the restart.